### PR TITLE
Add PHPStan to CI

### PR DESCRIPTION
**Pull request by mvorisek**

### CODE REVIEW

**Comment by mvorisek** [comment]
@ibelar @DarkSide666 level 2 is very mild, please look at the errors and please fix

File: .github/workflows/test-unit.yml [hunk 1]
```diff
@@ -46,8 +48,9 @@ jobs:
 
       - name: Install PHP dependencies
         run: |
-          if [ "${{ matrix.type }}" != "Phpunit" ]; then composer remove --no-interaction --no-update phpunit/phpunit johnkary/phpunit-speedtrap phpunit/phpcov --dev ; fi
+          if [ "${{ matrix.type }}" != "Phpunit" ] && [ "${{ matrix.type }}" != "StaticAnalysis" ]; then composer remove --no-interaction --no-update phpunit/phpunit johnkary/phpunit-speedtrap phpunit/phpcov --dev ; fi
```
Comment (mvorisek) [inline .github/workflows/test-unit.yml:51]: phpunit & behat deps are needed to correctly analyse tests and behat/Context class...

File: phpstan.neon.dist [hunk 2]
```diff
@@ -0,0 +1,8 @@
+parameters:
+    level: 2
```
Comment (mvorisek) [inline phpstan.neon.dist:2]: ideal is level 6

File: src/Form/Control/ScopeBuilder.php [hunk 3]
```diff
@@ -205,8 +205,6 @@ class ScopeBuilder extends Control
             self::OPERATOR_TEXT_DOESNOT_BEGIN_WITH => Condition::OPERATOR_NOT_LIKE,
             self::OPERATOR_TEXT_ENDS_WITH => Condition::OPERATOR_LIKE,
             self::OPERATOR_TEXT_DOESNOT_END_WITH => Condition::OPERATOR_NOT_LIKE,
-            self::OPERATOR_EQUALS => Condition::OPERATOR_EQUALS,
```
Comment (mvorisek) [inline src/Form/Control/ScopeBuilder.php:208]: @georgehristov @ibelar correct to remove? (keys were duplicate)

Comment (georgehristov) [reply src/Form/Control/ScopeBuilder.php:208]: Where is the duplicate?

Comment (mvorisek) [reply src/Form/Control/ScopeBuilder.php:208]: `self::OPERATOR_EQUALS` evaluates to same value as one OPERATOR_TEXT_... :)

Comment (georgehristov) [reply src/Form/Control/ScopeBuilder.php:208]: I believe not:
```
protected const OPERATOR_TEXT_EQUALS = 'is exactly';
...
protected const OPERATOR_EQUALS = 'equals';
```

Comment (mvorisek) [reply src/Form/Control/ScopeBuilder.php:208]: at least `OPERATOR_TEXT_DOESNOT_EQUAL` and `OPERATOR_DOESNOT_EQUAL`

File: phpstan.neon.dist [hunk 4]
```diff
@@ -0,0 +1,330 @@
...
+        - build/
+        - vendor/
+
+    # TODO review once we drop PHP 7.x support
+    treatPhpDocTypesAsCertain: false
+
+    ignoreErrors:
+        - '~^Unsafe usage of new static\(\)\.$~'
+
+        # TODO these rules are generated, this ignores should be fixed in the code
```
Comment (mvorisek) [inline phpstan.neon.dist:16]: we will merge now and every newly introdues error is checked

any help (later, in another PR) is welcomed

File: src/Form/Control/ScopeBuilder.php (hunk 3 above)
Comment (mvorisek) [reply src/Form/Control/ScopeBuilder.php:208]: see https://github.com/atk4/ui/runs/1486496532

**Review by DarkSide666 — APPROVED** [review]
LGTM

**Pull request merged**